packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionContainerStartCommand.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Uri } from 'vscode';
+import {
+  FilePathGatherer,
+  SfdxCommandlet,
+  SfdxWorkspaceChecker
+} from '../util';
+import {
+  ForceFunctionContainerStartExecutor,
+  validateStartFunctionsUri
+} from './forceFunctionStart';
+
+export const CONTAINER_START_TEXT_KEY = 'force_function_container_start_text';
+export const FUNCTION_CONTAINER_LOG_NAME = 'force_function_container_start';
+
+/**
+ * Executes sfdx run:function:start --verbose
+ * @param sourceUri
+ */
+export const forceFunctionContainerStartCommand = async (sourceUri?: Uri) => {
+  const validSourceUri = validateStartFunctionsUri(sourceUri);
+  if (validSourceUri === undefined) {
```

Comment:
I recommend doing `if (!validSourceUri)` over `if (validSourceUri === undefined)`.  This way, if it's null or undefined it will return.  Yes, the TS says it returns undefined (and not undefined | null), but I've seen cases where undefined was expected, but one got  null instead.

Reply:
def please call that out if you run into it again.  If we can't trust the typing of typescript then there's def a problem (probably :any fun). 
So the condition I was a little concerned about was '' (emptystring). 
The way the validate function works it should default to the activeTextEditor Uri, but I wasn't 100% clear at the time I touched this.  I'm fine with making the change though b/c I prefer that syntax as well.